app/src/main/java/de/westnordost/streetcomplete/quests/address/AddAddressStreet.kt

```diff
@@ -0,0 +1,108 @@
+        return true
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox) =
+            bbox.toGlobalOverpassBBox() + """
+            relation["type"="associatedStreet"];
+            > -> .inStreetRelation;
+
+            way["highway" ~ "^(${OsmTaggings.ALL_ROADS.joinToString("|")})$"]["name"] -> .named_roads;
+
+            nwr["addr:street"!~".*"]["addr:housenumber"]["addr:place"!~".*"](around.named_roads:150) -> .missing_data;
+
+            (.missing_data; - .inStreetRelation;);""" +
+                    getQuestPrintStatement()
+
+    // TODO: probably either this should be modified to match AddRoadName with custom set of roads  with config array migrated to OsmTaggings
+    // TODO: or AddRoadName can use something like that
+    /** return overpass query string to get roads with names
+     * */
+    private fun getStreetNameSuggestionsOverpassQuery(bbox: BoundingBox) =
+            bbox.toGlobalOverpassBBox() + "\n" + """
+        way[highway ~ "^(${OsmTaggings.ALL_ROADS.joinToString("|")})$"][name];
+        out body geom;""".trimIndent()
+
+    override fun applyAnswerTo(answer: AddressStreetAnswer, changes: StringMapChangesBuilder) {
+        var key = ""
+        when(answer){
+            is StreetName -> {key = "addr:street"}
+            is PlaceName -> {key = "addr:place"}
+        }
+        //TODO: can we actually get addr:street/place:langcode here, and if so,
+        // should we polute OSM with this?
+        for ((languageCode, name) in answer.localizedNames) {
+            if (languageCode.isEmpty()) {
+                changes.addOrModify(key, name)
+            } else {
+                changes.addOrModify("$key:$languageCode", name)
+            }
+        }
+    }
+
+    private fun putRoadNameSuggestion(element: Element, geometry: ElementGeometry?) {
+        if (element.type != Element.Type.WAY) return
+        if (geometry !is ElementPolylinesGeometry) return
+        val namesByLanguage = element.tags?.toRoadNameByLanguage() ?: return
+
+        roadNameSuggestionsDao.putRoad(element.id, namesByLanguage, geometry.polylines.first())
+    }
+
+    companion object {
+        //TODO: this only appears to pick up nodes, so street-name suggestions don't appear
+        // in the middle of a long straight-stretch. There might need to be a better way.
+        const val MAX_DIST_FOR_ROAD_NAME_SUGGESTION_IN_METERS = 250.0
+    }
+}
+
+/** OSM tags (i.e. name:de=Bäckergang) to map of language code -> name (i.e. de=Bäckergang) */
+private fun Map<String,String>.toRoadNameByLanguage(): Map<String, String>? {
+    val result = mutableMapOf<String,String>()
+    val namePattern = Pattern.compile("name(:(.*))?")
+    for ((key, value) in this) {
+        val m = namePattern.matcher(key)
+        if (m.matches()) {
+            val languageCode = m.group(2) ?: ""
+            result[languageCode] = value
+        }
+    }
+    return if (result.isEmpty()) null else result
+}
```

Comment:
Copypasta from `AddRoadName`.
Suggestion: Create an extension function in a new file `app/src/main/java/de/westnordost/streetcomplete/quests/address/RoadNameSuggestionsDao.kt` called
`fun RoadNameSuggestionsDao.putOsmWayAsRoad(way: Way, wayGeometry: ElementPolylinesGeometry)`
or similar which is used by both classes.

Reply:
Done in https://github.com/westnordost/StreetComplete/pull/1782/commits/5591b1a4e4c4a91d3591fdefcb8b74543899ef84